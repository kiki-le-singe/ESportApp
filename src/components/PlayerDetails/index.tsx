import React from 'react';
import { View, Image, Text } from 'react-native';
import PlayerInterface from 'src/interfaces/Player';
import { gramsToKilograms, total } from 'src/utils/';

interface Props {
  player: PlayerInterface;
}

export default class PlayerDetails extends React.PureComponent<Props> {
  render() {
    const { picture, data } = this.props.player;
    const { height, weight, age, last } = data;

    return (
      <View>
        <Image style={{ width: 125, height: 200 }} source={{ uri: picture }} />
        <Text>height: {height} cm</Text>
        <Text>age: {age}</Text>
        <Text>weight: {gramsToKilograms(weight)} kg</Text>
        <Text>total wins: {total(last, 1)}</Text>
        <Text>total losses: {total(last, 0)}</Text>
      </View>
    );
  }
}
