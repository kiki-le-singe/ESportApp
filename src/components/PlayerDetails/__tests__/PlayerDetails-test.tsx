import React from 'react';
import renderer from 'react-test-renderer';
import PlayerDetails from '../';

const player = {
  firstname: 'Foo',
  lastname: 'Bar',
  shortname: 'FooBar',
  picture: 'https://i.eurosport.com/_iss_/person/pp_clubteam/large/435121.jpg',
  country: {
    picture: 'https://i.eurosport.com/_iss_/geo/country/flag/large/2203.png'
  },
  data: {
     weight: 81000,
     height: 183,
     age: 33,
     last: [
       1,
       1,
       1,
       0,
       1
     ]
   }
}

it('renders correctly with defaults', () => {
  const button = renderer
    .create(<PlayerDetails player={player} />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
