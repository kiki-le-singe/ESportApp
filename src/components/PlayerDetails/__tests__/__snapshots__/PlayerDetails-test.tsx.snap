// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View>
  <Image
    source={
      Object {
        "uri": "https://i.eurosport.com/_iss_/person/pp_clubteam/large/435121.jpg",
      }
    }
    style={
      Object {
        "height": 200,
        "width": 125,
      }
    }
  />
  <Text>
    height: 
    183
     cm
  </Text>
  <Text>
    age: 
    33
  </Text>
  <Text>
    weight: 
    81
     kg
  </Text>
  <Text>
    total wins: 
    4
  </Text>
  <Text>
    total losses: 
    1
  </Text>
</View>
`;
